src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -0,0 +1,255 @@
+		visitsDashboardPage.waitForPageToLoad();
+		dashboardPage = visitsDashboardPage.goToPatientDashboard();
+	}
+
+	// User story: Add known allergies
+	@And("a user clicks on Allergies link from Patient dashboard page")
+	public void loadAllergiesPage() {
+		allergyPage = (AllergyPage) dashboardPage.clickOnAllergiesWidgetLink()
+				.waitForPage();
+	}
+
+	@Then("the system loads Allergies board page")
+	public void systemLoadsAllergiesPage() {
+		assertTrue(textExists("Allergies"));
+	}
+
+	@And("a user clicks Add Known Allergy button")
+	public void addKnownAllergy() {
+		addOrEditAllergyPage = allergyPage.clickOnAddNewAllergy();
+		addOrEditAllergyPage.enterDrug(DRUG_NAME);
+		addOrEditAllergyPage.drugId();
+		addOrEditAllergyPage.enterReaction(REACTION);
+		addOrEditAllergyPage.reactionId();
+		addOrEditAllergyPage.addAllergyNote("The effect is severe");
+	}
+
+	@And("a user clicks on save allergy button")
+	public void saveKnownAllergy() {
+		allergyPage = addOrEditAllergyPage.clickOnSaveAllergy();
+	}
+
+	@Then("the system adds known allergies into the allergies table")
+	public void systemAddsKnownAllergy() {
+		assertNotNull(addOrEditAllergyPage.getAllergiesList());
+		dashboardPage = addOrEditAllergyPage.clickReturn();
+	}
+
+	// User story: Add known condition
+	@And("a user clicks on Conditions link from Patient dashboard")
+	public void loadManageConditionsPage() {
+		conditionsPage = (ConditionsPage) dashboardPage
+				.clickOnConditionsWidgetLink().waitForPage();
+	}
+
+	@Then("the system loads Manage Conditions Page")
+	public void systemLoadsManageConditionsPage() {
+		assertTrue(conditionsPage.containsText("Conditions"));
+	}
+
+	@And("a user clicks on Add new condition")
+	public void userClicksAddNewCondition() {
+		conditionPage = (ConditionPage) conditionsPage.clickOnAddNewCondition()
+				.waitForPage();
+	}
+
+	@And("a user enters patient condition")
+	public void enterExistingCondition() {
+		conditionPage.typeInCondition(CONDITION);
+	}
+
+	@And("a user clicks save button")
+	public void saveCondition() {
+		conditionPage.clickSave();
+	}
+
+	@Then("the system adds New Condition in Conditions table")
+	public void systemAddsCondition() {
+		assertNotNull(conditionsPage.getConditionsList());
+		dashboardPage = conditionsPage.clickReturn();
+	}
+
+	// User story: Attach supporting document
+	@And("a user clicks on Attachments link from patient visits dashboard")
+	public void loadAttachmentsPage() {
+		visitsDashboardPage = (PatientVisitsDashboardPage) dashboardPage
+				.goToRecentVisits();
+		attachmentsPage = (AttachmentsPage) dashboardPage.goToAttachmentsPage()
+				.waitForPage();
+	}
+
+	@Then("the system loads Attachments page")
+	public void systemloadsAttachmentsPage() {
+		assertTrue(textExists("Attachments"));
+	}
+
+	@And("a user attaches patient supporting file")
+	public void addSupportingFile() {
+		attachmentsPage.setFileUrl("/home/opensource/Documents/Form.pdf");
```

Comment:
The method `setFileUrl()` will be refactored when this [issue](https://issues.openmrs.org/browse/RATEST-164) is resolved !